User Interface Text Guidelines

HilaryClarke opened the pull request:
@justinegeffen  I'm having trouble inserting images into the table. 
cc @andrewbrown00 


#### Summary
<!--
A description of what this pull request does.
-->

HilaryClarke committed:
Create User Interface Text Guidelines.md

Migrating https://docs.mattermost.com/process/documentation-UItext-guidelines.html to new Handbook. 

@andrewbrown00 to review

HilaryClarke committed:
Add files via upload

HilaryClarke committed:
Update User Interface Text Guidelines.md

HilaryClarke committed:
Add files via upload

HilaryClarke committed:
Update User Interface Text Guidelines.md

jasonblais commented:
/check-cla

justinegeffen commented:
Thanks for this PR, @HilaryClarke! Quick question around it: Is this from the existing content on docs.mattermost.com? I know that @andrewbrown00 may have included this in the UX style guide so it might be an option to link there and avoid content duplication. :)

jasonblais commented:
@justinegeffen You're correct, it is from https://docs.mattermost.com/process/documentation-UItext-guidelines.html, and it's a separate page from the UX style guide.

justinegeffen commented:
Thanks, @jasonblais. I'm open to adding this to the Handbook. The initial idea discussed with the UX team was to include their Compass Design guide in the overall style guide, which includes the UX writing guidelines. 

An alternative is to bring that content back over to the handbook and then link out to the Compass Design guide. It is slightly updated compared to this version IIRC. 

@HilaryClarke, the guide is [here](https://docs.google.com/document/d/1YjyGHXdcPJyamW6HPjZbVP5T2u07AJf5QvluwDNQneA/edit#heading=h.5vqelzivnmzb). Let me know if you'd like to copy that over otherwise I'm happy to. :)

jasonblais commented:
@justinegeffen That is great!

For the purposes of this (which is moving the legacy handbook over to handbook), I would recommend adding the page. The UX team can then incorporate the UI text guidelines to the Compass Design guide and apply the changes when the feel the guide is ready.

justinegeffen commented:
Great, thanks @jasonblais. Apologies for the delayed response. I'll work on this PR today.

justinegeffen committed:
Update and rename User Interface Text Guidelines.md to user-interface-text-guidelines.md

Replaced content with content from the Compass Design System doc for consistency.

HilaryClarke commented:
Thanks @justinegeffen  Let me know if you need anything from me!

jasonblais commented:
@justinegeffen I was actually thinking of also keeping the current UI text guidelines that Hilary submitted until UX design has decided how to incorporate them into the new guide. Sorry that wasn't clear.

justinegeffen commented:
Oh! My bad, I should have clarified that before going ahead. 

The first set of content (preceding the guidelines around buttons, labels etc) should be the same with possible minor wording changes. I'll review that today. I can remove the content around labels and buttons and re-add the table today. :)

jasonblais commented:
Ah! You're right, it should be okay to keep the content around labels and buttons. But the table may be a useful reference point.

justinegeffen committed:
Update user-interface-text-guidelines.md

Table re-added.

justinegeffen commented:
I'll sort out the images in the table shortly.

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+## UX Writing Principles
+
+Following the Writing Guidelines below will ensure communications to our users feel like they are coming from the same voice.
```
```suggestion
Following these guidelines ensures that in-product communication to our users has the the same, consistent, voice.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+# User Interface Text Guidelines
+
+## UX Writing Principles
```
```suggestion
## UX writing principles
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+  - If the subject (the doer of an action) is the Mattermost application itself. For example, it’s okay to say **The image is deleted** instead of **The server deleted the image**.
+- If you’re asking the user to do or not do something, use imperatives (command phrases). For example, say **Do not change the Hostname** instead of saying **It is not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example, say **Do not change the Hostname because if you do so <this bad thing will happen>**.
+
+### Guidelines for System Messages
```
```suggestion
### Guidelines for system messages
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+### Guidelines for System Messages
+
+Don’t blame the user. Inform them about what happened, explain why it happened, and suggest a way forward. Try to use complete sentences in your messages. A sentence phrase (an incomplete sentence) might sound okay in English but could present internationalization challenges.
```
```suggestion
Don’t blame the user. Inform them about what happened, explain why it happened, and suggest a way forward. Try to use complete sentences in your messages. A sentence phrase (an incomplete sentence) might make sense in English but could present internationalization challenges.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+Don’t blame the user. Inform them about what happened, explain why it happened, and suggest a way forward. Try to use complete sentences in your messages. A sentence phrase (an incomplete sentence) might sound okay in English but could present internationalization challenges.
+
+If a System Message contains variables (tokens):
```
```suggestion
If a system message contains variables (tokens):
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+Don’t blame the user. Inform them about what happened, explain why it happened, and suggest a way forward. Try to use complete sentences in your messages. A sentence phrase (an incomplete sentence) might sound okay in English but could present internationalization challenges.
+
+If a System Message contains variables (tokens):
+- Do not use verbs or adjectives as variables.
```
```suggestion
- Don't use verbs or adjectives as variables.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+If a System Message contains variables (tokens):
+- Do not use verbs or adjectives as variables.
+- Do not create plurals of variables by adding an s.
```
```suggestion
- Don't create plurals of variables by adding an "s".
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+If a System Message contains variables (tokens):
+- Do not use verbs or adjectives as variables.
+- Do not create plurals of variables by adding an s.
+- If the variable is a noun, use a qualifier after the variable. For example, say **The {channel_name} channel was created** instead of saying **The {channel_name} was created**.
```
```suggestion
- If the variable is a noun, use a qualifier after the variable. For example, say **The {channel_name} channel was created** rather than **The {channel_name} was created**.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- Do not create plurals of variables by adding an s.
+- If the variable is a noun, use a qualifier after the variable. For example, say **The {channel_name} channel was created** instead of saying **The {channel_name} was created**.
+
+System Messages can be of the following types: **notification**, **confirmation**, **warning**, and **error**. The following sections contain guidelines that are specific to each of these types.
```
```suggestion
System messages can be of the following types: **notification**, **confirmation**, **warning**, and **error**. The following sections contain guidelines that are specific to each of these types.
```

justinegeffen replied:
```suggestion
System Messages can be one of the following types: **notification**, **confirmation**, **warning**, and **error**. The following sections contain guidelines that are specific to each of these types.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+### Notification messages
+
+A notification message is for informing somebody about something that happened. Such messages do not need any user input, and do not prevent a user from continuing to use Mattermost.
```
```suggestion
A notification message informs the user about an event or action that took place. These messages don't need any user input, and don't prevent the user from continuing to use Mattermost.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+### Confirmation messages
+
+A confirmation message is for asking somebody to confirm whether the immediately preceding command should be proceeded with. The user cannot use Mattermost until a confirmation is given or denied.
```
```suggestion
A confirmation message requires user input to confirm that they want to proceed with the action. Confirmation must be provided (either to continue or cancel) before the user can continue to use Mattermost.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+### Warning messages
+
+A warning message is for alerting somebody about something that might go wrong. The user can continue using Mattermost unless the warning message needs an explicit user input.
```
```suggestion
A warning message alerts the user that something that might go wrong. They can continue using Mattermost unless the warning message needs user input.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+A warning message is for alerting somebody about something that might go wrong. The user can continue using Mattermost unless the warning message needs an explicit user input.
+
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
```
```suggestion
- Explain what has happened, or can happen, and what may go wrong as a result.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+### Error messages
+
+An error message is for telling somebody that something went wrong. Errors prevent a user from doing a task or accessing a feature till the error is resolved.
```
```suggestion
An error message informs the user that something went wrong. Errors prevent the user from completing a task or accessing a feature until the error is resolved.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- If a solution or workaround isn’t obvious, suggest one.
+- Examples:
+  - **Messages must have fewer than 120 characters.**
+  - **A connection to the marketplace server could not be established. Check your settings on the [System Console](<hyperlink>)**. 
```
```suggestion
  - **A connection to the Plugin Marketplace server could not be established. Check your settings in the [System Console](<hyperlink>)**. 
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Titles** 
+
+Titles should use Sentence case. Exceptions would be when proper nouns are being used.
```
```suggestion
Titles should use sentence case (e.g. **This article is about user interface content**) with the exception of proper nouns (e.g. **This article is about Mattermost**).
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+Titles should use Sentence case. Exceptions would be when proper nouns are being used.
+
+**Button labels, input labels and navigation labels** 
```
```suggestion
**Button labels, input labels, and navigation labels** 
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Button labels, input labels and navigation labels** 
+
+These labels should use Sentence case. Exceptions would be when proper nouns are being used.
```
```suggestion
These labels should use sentence case. (e.g. **This article is about user interface content**) with the exception of proper nouns (e.g. **This article is about Mattermost**).
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Titles**
+
+Titles should omit periods unless the headline is more than one sentence.
```
```suggestion
Titles shouldn't have periods unless the headline is more than one sentence.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Paragraphs**
+
+Paragraphs should always use proper punctuation.
```
```suggestion
Paragraphs should always be properly punctuated.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Bullets**
+
+Bullets should omit periods unless the bullet text is more than one sentence.
```
```suggestion
Bullets lists shouldn't have periods unless the bullet text is more than one sentence.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Button labels**
+
+Button labels should always omit periods or punctuation.
```
```suggestion
Button labels shouldn't have periods or other punctuation.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+<td>Headline style</td>
+<td><ul>
+<li>Verb or verb phrase</li>
+<li>No longer than 3 words</li>
```
```suggestion
<li>No longer than three words</li>
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+<td><ul>
+<li>Noun, verb, or sentence fragment</li>
+<li>No punctuation</li>
+<li>Brief (lengthier explanations belong to help text)</li>
```
```suggestion
<li>Brief (longer explanations belong to help text)</li>
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+</tbody>
+</table>
+
+.. [*] For headline style, capitalize all words except those with 3 letters or fewer, articles (*a*, *an*, *the*), prepositions (*on*, *to*, *in*, *from*, *of*), and coordinating conjunctions (*and*, *but*, *or*, *for*). Despite these exceptions, always capitalize the first and last word. For sentence style, capitalize only the first word.
```
```suggestion
.. [*] For headline style, capitalize all words except those with three letters or fewer, articles (*a*, *an*, *the*), prepositions (*on*, *to*, *in*, *from*, *of*), and coordinating conjunctions (*and*, *but*, *or*, *for*). Despite these exceptions, always capitalize the first and last word. For sentence case, capitalize only the first word.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Concise**
+
+Use short words and sentences. Avoid unnecessary fluff.
```
```suggestion
Use short words and sentences.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+
+**Clear**
+
+Always write for clarity and simplicity. Err on the side of simple language that can be understood by novice users and avoid vague terms.
```
```suggestion
Always write for clarity and simplicity. Err on the side of simple language that can be understood by novice users, and avoid vague terms and jargon.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- Use the present tense to describe a current state or condition, and the future tense to state something that is very definitely going to happen.
+- Use the active voice, except for these cases:
+  - If you’ll end up blaming the user. For example, don’t say **You entered an incorrect password**. Instead, say **The password is incorrect**.
+  - If you’re describing what just happened. For example, it’s okay to say **Your incoming webhook is set up**.
```
```suggestion
  - If you’re describing what just happened. For example: **Your incoming webhook is set up**.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- Use the active voice, except for these cases:
+  - If you’ll end up blaming the user. For example, don’t say **You entered an incorrect password**. Instead, say **The password is incorrect**.
+  - If you’re describing what just happened. For example, it’s okay to say **Your incoming webhook is set up**.
+  - If the subject (the doer of an action) is the Mattermost application itself. For example, it’s okay to say **The image is deleted** instead of **The server deleted the image**.
```
```suggestion
  - If the subject (the doer of an action) is the Mattermost application itself. For example: **The image has been deleted** instead of **The server deleted the image**.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+  - If you’ll end up blaming the user. For example, don’t say **You entered an incorrect password**. Instead, say **The password is incorrect**.
+  - If you’re describing what just happened. For example, it’s okay to say **Your incoming webhook is set up**.
+  - If the subject (the doer of an action) is the Mattermost application itself. For example, it’s okay to say **The image is deleted** instead of **The server deleted the image**.
+- If you’re asking the user to do or not do something, use imperatives (command phrases). For example, say **Do not change the Hostname** instead of saying **It is not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example, say **Do not change the Hostname because if you do so <this bad thing will happen>**.
```
```suggestion
- If you’re asking the user to do or not do something, use imperatives (command phrases). For example, use **Don't change the Hostname** instead of **It's not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example: **Don't change the Hostname because doing so <could cause this bad thing will happen>**.
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- Examples:
+  - **The Enterprise license expires in 2 days. If you do not renew it, some features will be disabled on licence expiry.**
```
```suggestion
  - **The Enterprise license expires in 2 days. If it's not renewed, some features will be disabled on license expiry.**
```

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- Examples:
+  - **The Enterprise license expires in 2 days. If you do not renew it, some features will be disabled on licence expiry.**
+  - **If you claim this AD/LDAP account, you will no longer be able to log in with your email.**
```
```suggestion
  - **If you claim this AD/LDAP account, you won't be able to log in with your email address.**
```

justinegeffen reviewed (commented):
I've provided some feedback and re-added the table. I'm looking into the images now.

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+</tr>
+<tr class="even">
+<td>Tooltip</td>
+<td>![](../../../.gitbook/assets/tooltipimage.png)</td>
```
```suggestion
<td>![](../../../.gitbook/assets/Tooltipimage.png)</td>
```

justinegeffen committed:
Update operations/research-and-development/product/user-interface-text-guidelines.md

justinegeffen committed:
Update operations/research-and-development/product/user-interface-text-guidelines.md

mattermod commented:
Thanks @HilaryClarke for the pull request!

 Per the [**CONTRIBUTING.md**](https://github.com/mattermost/platform/blob/master/CONTRIBUTING.md#submitting-a-pull-request) file displayed when you created this pull request, we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**

This is a standard procedure for many open source projects. Your form should be processed within 24 hours and reviewers for your pull request will be able to proceed.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

justinegeffen committed:
Update operations/research-and-development/product/user-interface-text-guidelines.md

justinegeffen committed:
Apply suggestions from code review

justinegeffen commented on operations/research-and-development/product/user-interface-text-guidelines.md:
```diff
@@ -0,0 +1,266 @@
+- Don't create plurals of variables by adding an "s".
+- If the variable is a noun, use a qualifier after the variable. For example, say **The {channel_name} channel was created** rather than **The {channel_name} was created**.
+
+System messages can be of the following types: **notification**, **confirmation**, **warning**, and **error**. The following sections contain guidelines that are specific to each of these types.
```
```suggestion
System messages can be one of the following types: **notification**, **confirmation**, **warning**, and **error**. The following sections contain guidelines that are specific to each of these types.
```

justinegeffen committed:
Update operations/research-and-development/product/user-interface-text-guidelines.md

justinegeffen commented:
I still can't get the images to display in the table. But they display when they're outside of the table (i.e., I move the link to another place in the doc). Next option is to change the table formatting.

jasonblais committed:
Fix image formatting, with https://stackoverflow.com/a/61784060

jasonblais commented:
@justinegeffen I found a StackOverflow article with a tip to use `<img>` tags inside the table: https://stackoverflow.com/a/61784060

I did so and it seems to work - except the dimensions are really small. Would you be open to help take a look at making the dimensions reasonable in size?

justinegeffen commented:
I've taken a look at the table in the preview and the images look like they're the right size. I can definitely resize them though. Just need to find the right sizing. LMKWYT. 

![Screen Shot 2020-06-28 at 20 35 11](https://user-images.githubusercontent.com/4223530/85955454-fbd0bb80-b97e-11ea-8eee-8721456f8fda.png)
![Screen Shot 2020-06-28 at 20 35 20](https://user-images.githubusercontent.com/4223530/85955455-fd9a7f00-b97e-11ea-849e-b6f54f7a837b.png)

jasonblais reviewed (approved):
Thank you @justinegeffen, looks great from my end :+1:
